compiler/environment.ml

```diff
@@ -0,0 +1,116 @@
+open Ast;;
+
+module FunctionMap = Map.Make(String);;
+module VariableMap = Map.Make(String);;
+module ArrayTypeMap = Map.Make(String);;
+
+exception VarAlreadyDeclared;;
+exception VarNotDeclared;;
+exception FunctionAlreadyDeclared;;
+exception FunctionNotDeclared;;
+exception IncorrectFunctionParameterTypes;;
+exception MixedTypeArray;;
+exception ArrayLiteralTypeMismatch;;
+
+type func_info  = {
+  id : string; 
+  return : data_type; 
+  args : data_type list;
+  arg_names: string list;
+}
+
+type symbol_table = {
+  func_map: func_info FunctionMap.t;
+  var_map: data_type VariableMap.t;
+  array_type_map: data_type ArrayTypeMap.t;
+}
+
+let create = 
+  {
+    func_map = FunctionMap.empty;
+    var_map = VariableMap.empty;
+    array_type_map = ArrayTypeMap.empty;
+  }
+
+let update f_map v_map a_type_map =
```

Comment:
This is how I saw a few other groups doing it - it seems like a nice way to be able to update the map. Especially since we're not using global variables or anything (we're just passing a new env back after each statement), this seemed like a good way to do it